hollow/src/main/java/com/netflix/hollow/api/consumer/incubating/HollowConsumerV2.java

```diff
@@ -0,0 +1,515 @@
+            requireNonNull(blobRetriever, "blobRetriever cannot be null; use blobRetrieverFeature().disable()");
+            return this.blobRetriever.set(blobRetriever);
+        }
+
+        public Feature<BlobRetriever> blobRetrieverFeature() {
+            return blobRetriever;
+        }
+
+        @Deprecated
+        public synchronized Builder<A> withLocalBlobStore(File localBlobStore) {
+            return withLocalBlobStore(localBlobStore.toPath());
+        }
+
+        public synchronized Builder<A> withLocalBlobStore(Path localBlobStore) {
+            requireNonNull(localBlobStore, "localBlobStore cannot be null; use localBlobStoreFeature().disable()");
+            return this.localBlobStore.set(localBlobStore);
+        }
+
+        public Feature<Path> localBlobStoreFeature() {
+            return localBlobStore;
+        }
+
+        public synchronized Builder<A> withAnnouncementWatcher(AnnouncementWatcher announcementWatcher) {
+            requireNonNull(announcementWatcher, "announcementWatcher cannot be null; use announcementWatcherFeature().disable()");
+            return this.announcementWatcher.set(announcementWatcher);
+        }
+
+        public Feature<AnnouncementWatcher> announcementWatcherFeature() {
+            return announcementWatcher;
+        }
+
+        public synchronized Builder<A> withRefreshListener(RefreshListener refreshListener) {
+            requireNonNull(announcementWatcher, "refreshListener cannot be null");
+            Set<RefreshListener> list = refreshListeners.hasValue()
+                    ? refreshListeners.get()
+                    : new LinkedHashSet<>();
+            list.add(refreshListener);
+            return refreshListeners.set(list);
+        }
+
+        public synchronized Builder<A> withRefreshListeners(RefreshListener... additionalListeners) {
+            requireNonNull(announcementWatcher, "additionalListeners cannot be null");
+            Set<RefreshListener> l = refreshListeners.hasValue()
+                    ? refreshListeners.get()
+                    : new LinkedHashSet<>();
+            Collections.addAll(l, additionalListeners);
+            return refreshListeners.set(l);
+        }
+
+        public Feature<Set<RefreshListener>> refreshListenersFeature() {
+            return refreshListeners;
+        }
+
+        /**
+         * Provide the code generated API class that extends {@link HollowAPI}.
+         *
+         * The instance returned from {@link HollowConsumer#getAPI()} will be of the provided type and can be cast
+         * to access generated methods.
+         *
+         * @param apiClass the code generated API class
+         * @return this builder
+         * @throws IllegalArgumentException if provided API class is {@code HollowAPI} instead of a subclass
+         */
+        public synchronized <A2 extends A> Builder<A2> withAPI(Class<A2> apiClass) {
+            /*
+             * FIXME(timt): not great...
```

Comment:
Users are unlikely to come across this wart when using the builder in a fluent style, but I don't like it nonetheless.

Maybe we make it an error to call `withAPI(...)` more than once with a different `HollowAPI` subclass?

I like the solutions I've come up with about as much as leaving this as-is. Which is to say, I don't like any option that much.